tests/benchmarks/scimark2_benchmark_test.py

```diff
@@ -0,0 +1,104 @@
+Using       2.00 seconds min time per kenel.
+Composite Score:         1596.04
+FFT             Mflops:  1568.64    (N=1024)
+SOR             Mflops:  1039.98    (100 x 100)
+MonteCarlo:     Mflops:   497.64
+Sparse matmult  Mflops:  1974.39    (N=1000, nz=5000)
+LU              Mflops:  2899.56    (M=100, N=100)
+"""
+
+TEST_OUTPUT_JAVA = """;;; Java small
+SciMark 2.0a
+
+Composite Score: 1716.3662351463677
+FFT (1024): 1000.1380057152871
+SOR (100x100):   1353.1987180103354
+Monte Carlo : 727.7138820888014
+Sparse matmult (N=1000, nz=5000): 1495.40225150659
+LU (100x100): 4005.3783184108247
+
+java.vendor: Oracle Corporation
+java.version: 1.7.0_75
+os.arch: amd64
+os.name: Linux
+os.version: 3.16.0-25-generic
+"""
+
+EXPECTED_C_METADATA = {
+    'bench_version': 'C small',
+}
+
+EXPECTED_JAVA_METADATA = {
+    'bench_version': 'Java small',
+    'java.vendor': 'Oracle Corporation',
+    'os.version': '3.16.0-25-generic',
+    'os.arch': 'amd64',
+    'os.name': 'Linux',
+    'java.version': '1.7.0_75',
+}
+
+EXPECTED_RESULT = [
+    sample.Sample(metric='Composite Score', value=1596.04,
+                  unit='Mflops', metadata=EXPECTED_C_METADATA),
+    sample.Sample(metric='FFT (N=1024)', value=1568.64,
+                  unit='Mflops', metadata=EXPECTED_C_METADATA),
+    sample.Sample(metric='SOR (100 x 100)', value=1039.98,
+                  unit='Mflops', metadata=EXPECTED_C_METADATA),
+    sample.Sample(metric='MonteCarlo', value=497.64,
+                  unit='Mflops', metadata=EXPECTED_C_METADATA),
+    sample.Sample(metric='Sparse matmult (N=1000, nz=5000)', value=1974.39,
+                  unit='Mflops', metadata=EXPECTED_C_METADATA),
+    sample.Sample(metric='LU (M=100, N=100)', value=2899.56,
+                  unit='Mflops', metadata=EXPECTED_C_METADATA),
+    sample.Sample(metric='Composite Score', value=1716.3662351463677,
+                  unit='Mflops', metadata=EXPECTED_JAVA_METADATA),
+    sample.Sample(metric='FFT (1024)', value=1000.1380057152871,
+                  unit='Mflops', metadata=EXPECTED_JAVA_METADATA),
+    sample.Sample(metric='SOR (100x100)', value=1353.1987180103354,
+                  unit='Mflops', metadata=EXPECTED_JAVA_METADATA),
+    sample.Sample(metric='Monte Carlo', value=727.7138820888014,
+                  unit='Mflops', metadata=EXPECTED_JAVA_METADATA),
+    sample.Sample(metric='Sparse matmult (N=1000, nz=5000)',
+                  value=1495.40225150659, unit='Mflops',
+                  metadata=EXPECTED_JAVA_METADATA),
+    sample.Sample(metric='LU (100x100)', value=4005.3783184108247,
+                  unit='Mflops', metadata=EXPECTED_JAVA_METADATA),
+]
+
+
+class Scimark2BenchmarkTestCase(unittest.TestCase):
+
+  def testParseResults(self):
+    samples = []
+    samples.extend(scimark2_benchmark.ParseResults(TEST_OUTPUT_C))
+    samples.extend(scimark2_benchmark.ParseResults(TEST_OUTPUT_JAVA))
+
+    self.assertEqual(samples, EXPECTED_RESULT)
```

Comment:
Consider splitting this into two methods so that it's clear from the test logs which parsing method (C, Java, or both) failed.